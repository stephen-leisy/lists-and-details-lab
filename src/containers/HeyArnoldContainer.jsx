import React, { Component } from 'react';
import { getCharacters } from '../services/heyArnoldApi';
import CharacterList from '../components/characters/CharacterList';

export default class HeyArnoldContainer extends Component {
  state = {
    characterList: [],
    loading: true,
  };

  async componentDidMount() {
    const characterList = await getCharacters();
    this.setState({ characterList, loading: false });
  }
  render() {
    const { characterList, loading } = this.state;

    if (loading === true) return <div>loading...</div>;

    return (
      <div className="main">
        <CharacterList list={characterList} />
      </div>
    );
  }
}
