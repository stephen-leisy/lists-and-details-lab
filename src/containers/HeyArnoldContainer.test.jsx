import React from 'react';
import { render, screen } from '@testing-library/react';
import HeyArnoldContainer from './HeyArnoldContainer';

describe('Hey Arnold Container Tests', () => {
  it.only('renders a list of characters to the page', async () => {
    render(<HeyArnoldContainer />);
    screen.getByText('loading...');
    const ul = await screen.findByRole('list', { name: 'characters' });
    expect(ul).toMatchSnapshot();
  });
});
