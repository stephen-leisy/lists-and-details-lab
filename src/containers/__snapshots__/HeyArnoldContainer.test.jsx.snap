// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hey Arnold Container Tests renders a list of characters to the page 1`] = `
<ul
  aria-label="characters"
>
  <a
    href="/5da237699734fdcb7bef8f51"
  >
    <li>
      <h1>
        Arnold Shortman
      </h1>
    </li>
  </a>
  <a
    href="/5da237699734fdcb7bef8f52"
  >
    <li>
      <h1>
        Grandpa Phil
      </h1>
    </li>
  </a>
  <a
    href="/5da237699734fdcb7bef8f53"
  >
    <li>
      <h1>
        Grandma Gertie
      </h1>
    </li>
  </a>
  <a
    href="/5da237699734fdcb7bef8f54"
  >
    <li>
      <h1>
        Mitzi
      </h1>
    </li>
  </a>
  <a
    href="/5da237699734fdcb7bef8f55"
  >
    <li>
      <h1>
        Arnie
      </h1>
    </li>
  </a>
  <a
    href="/5da237699734fdcb7bef8f56"
  >
    <li>
      <h1>
        Miles Shortman
      </h1>
    </li>
  </a>
  <a
    href="/5da237699734fdcb7bef8f57"
  >
    <li>
      <h1>
        Stella Shortman
      </h1>
    </li>
  </a>
  <a
    href="/5da237699734fdcb7bef8f58"
  >
    <li>
      <h1>
        Grandpa's father
      </h1>
    </li>
  </a>
  <a
    href="/5da237699734fdcb7bef8f59"
  >
    <li>
      <h1>
        Grandpa's grandfather
      </h1>
    </li>
  </a>
  <a
    href="/5da237699734fdcb7bef8f5a"
  >
    <li>
      <h1>
        Oskar Kokoshka
      </h1>
    </li>
  </a>
  <a
    href="/5da237699734fdcb7bef8f5b"
  >
    <li>
      <h1>
        Suzie Kokoshka
      </h1>
    </li>
  </a>
  <a
    href="/5da237699734fdcb7bef8f5c"
  >
    <li>
      <h1>
        Nancy
      </h1>
    </li>
  </a>
  <a
    href="/5da237699734fdcb7bef8f5d"
  >
    <li>
      <h1>
        Ernie Potts
      </h1>
    </li>
  </a>
  <a
    href="/5da237699734fdcb7bef8f5e"
  >
    <li>
      <h1>
        Mr. Hyunh
      </h1>
    </li>
  </a>
  <a
    href="/5da237699734fdcb7bef8f5f"
  >
    <li>
      <h1>
        Mai Hyunh
      </h1>
    </li>
  </a>
  <a
    href="/5da237699734fdcb7bef8f60"
  >
    <li>
      <h1>
        Mr. Smith
      </h1>
    </li>
  </a>
  <a
    href="/5da237699734fdcb7bef8f61"
  >
    <li>
      <h1>
        Lana Vail
      </h1>
    </li>
  </a>
  <a
    href="/5da237699734fdcb7bef8f62"
  >
    <li>
      <h1>
        Mr. Purdy (character)
      </h1>
    </li>
  </a>
  <a
    href="/5da237699734fdcb7bef8f63"
  >
    <li>
      <h1>
        Helga G. Pataki
      </h1>
    </li>
  </a>
  <a
    href="/5da237699734fdcb7bef8f64"
  >
    <li>
      <h1>
        Robert "Big Bob" Pataki
      </h1>
    </li>
  </a>
</ul>
`;
